File: core/src/main/java/com/cc/core/actions/ActionResult.java
package com.cc.core.actions;

public class ActionResult {
    public boolean success;
    public String message;
    public Object data;

    public ActionResult() {

    }

    public ActionResult(boolean success, String message) {
        this.success = success;
        this.message = message;
    }

    public static ActionResult successResult() {
        ActionResult result = new ActionResult();
        result.success = true;
        result.message = "ok";
        return result;
    }

    public static ActionResult failedResult(Throwable thr) {
        ActionResult result = new ActionResult();
        result.success = false;
        result.message = thr == null ? "unknown error" : thr.getClass().getName() + ":" + thr.getMessage();
        return result;
    }

    public static ActionResult failedResult(String message) {
        ActionResult result = new ActionResult();
        result.success = false;
        result.message = message;
        return result;
    }

    public boolean isSuccess() {
         return success;
    }

    @Override
    public String toString() {
        return "ActionResult{" +
                "success=" + success +
                ", message='" + message + '\'' +
                ", data=" + data +
                '}';
    }
}

File: core/src/main/java/com/cc/core/actions/ActionResult.kt
package com.cc.core.actions

import com.google.gson.annotations.SerializedName

class ActionResult {
	@SerializedName("success")
    var success: Boolean = false
	@SerializedName("message")
    var message: String? = null
	@SerializedName("id")
    var actionId: String = ""
	@SerializedName("data")
    var data: Any? = null

    private constructor()
    constructor(actionId : String, success: Boolean, message: String?) {
        this.success = success
        this.message = message
        this.actionId = actionId
    }

    fun isSuccess() : Boolean {
        return success
    }

    companion object {
        fun successResult(actionId : String): ActionResult {
            return successResult(actionId, null)
        }

        fun successResult(actionId : String, data: Any?): ActionResult {
            val result = ActionResult()
            result.actionId = actionId
            result.success = true
            result.message = "ok"
            result.data = data
            return result
        }


        fun failedResult(actionId : String, thr: Throwable?): ActionResult {
            val result = ActionResult()
            result.actionId = actionId
            result.success = false
            result.message = if (thr == null) "unknown error" else thr.javaClass.name + ":" + thr.message
            return result
        }

        fun failedResult(actionId : String, message: String): ActionResult {
            val result = ActionResult()
            result.success = false
            result.message = message
            result.actionId = actionId
            return result
        }
    }
}